java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -619,7 +622,10 @@ abstract class AbstractArcHost(
     open val arcHostContextCapability = Capabilities(Shareable(true))
 
     override suspend fun isHostForParticle(particle: Plan.Particle) =
-        registeredParticles().contains(ParticleIdentifier.from(particle.location))
+        if (particle.isExpressionParticle())
+            registeredParticles().contains(EvaluatorParticle::class.toParticleIdentifier())
```

Comment:
again, this wouldn't be needed if we just made a Plan.Particle that had location that mapped to EvaluatorParticle, and we pre-registered it in the ArcHost constructor.

Reply:
Also, we need to think about the multi-host scenario of where these things run. For now, leave it as be, but these should all probably run in PureHost/IsolatedHost at some point, and so should only be registered in that host eventually.

Reply:
Yup, reverted, no need for special casing - we will generate the explicit location in the recipe2plan.